File: Data Structures & Algorithms/max-area-of-island/submission-1.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int res = 0;
        int max = 0;
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == 1){
                    res = dfs(i,j, grid);
                    max = Math.max(max, res);


                }
                

            }
        }
        return max;
    }
    private int dfs(int i, int j, int[][] grid){
        if(i < 0 || i >= grid.length || j < 0 || j >= grid[0].length || grid[i][j] == 0){
            return 0;
        }
        grid[i][j] = 0;
        int count = 1;
        count += dfs(i+1, j, grid);
        count += dfs(i-1, j, grid);
        count += dfs(i, j+1, grid);
        count += dfs(i, j-1, grid);
        return count;

        
    }
}

File: Data Structures & Algorithms/count-number-of-islands/submission-1.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        count = 0
        row = len(grid)
        col = len(grid[0])
        #visited = [[False] * col for _ in range(row)]
        
        def dfs(r,c):
            if r < 0 or r >= row or c < 0 or c >= col or grid[r][c] == "0":
                return
            if grid[r][c] == "1":
                grid[r][c] = "0"

            #visited[r][c] = False
            dfs(r+1,c)
            dfs(r-1, c)
            dfs(r, c+1)
            dfs(r, c-1)
            
        
        for r in range(row):
            for c in range(col):
                if grid[r][c] == "1":
                    dfs(r,c)
                    count +=1
        return count